### PR TITLE
reuse memory-expensive tagger between all loaded English dialects

### PR DESCRIPTION
**Pull request by donnerpeter**

### CODE REVIEW

File: languagetool-language-modules/en/src/main/java/org/languagetool/language/English.java [hunk 1]
```diff
@@ -105,7 +107,12 @@ public String getShortCode() {
   @NotNull
   @Override
   public Tagger createDefaultTagger() {
-    return new EnglishTagger();
+    WeakReference<EnglishTagger> ref = cachedTagger;
+    EnglishTagger tagger = ref == null ? null : ref.get();
+    if (tagger == null) {
+      cachedTagger = new WeakReference<>(tagger = new EnglishTagger());
```
Comment (danielnaber) [inline languagetool-language-modules/en/src/main/java/org/languagetool/language/English.java:113]: Could these be split into two lines? I think it's a bit uncommon to have a second assignment like this.

Comment (donnerpeter) [reply languagetool-language-modules/en/src/main/java/org/languagetool/language/English.java:113]: It's common for me, but I see your point

Comment (donnerpeter) [reply languagetool-language-modules/en/src/main/java/org/languagetool/language/English.java]: Force-pushed

**Comment by donnerpeter** [comment]
Build failed with this. Unrelated to my change?

java.lang.IllegalArgumentException: Unexpected line format, expected at most one slash: Lösegeldversicherung #https://de.wikipedia.org/wiki/Entf%C3%BChrungsversicherung
	at com.google.common.cache.LocalCache$Segment.get(LocalCache.java:2051)
	at com.google.common.cache.LocalCache.get(LocalCache.java:3953)
	at com.google.common.cache.LocalCache.getOrLoad(LocalCache.java:3976)
	at com.google.common.cache.LocalCache$LocalLoadingCache.get(LocalCache.java:4960)
	at com.google.common.cache.LocalCache$LocalLoadingCache.getUnchecked(LocalCache.java:4966)
	at org.languagetool.rules.spelling.morfologik.MorfologikMultiSpeller.getPlainTextDictSpellerOrNull(MorfologikMultiSpeller.java:180)

**Pull request merged**